test/math.jl

```diff
@@ -1538,6 +1538,33 @@ end
             @test all((t -> ===(t...)), zip(x^y, p[y + 1]))
         end
     end
+    # issue #57464
+    @test Float32(1.1)^typemin(Int) == Float32(0.0)
+    @test Float16(1.1)^typemin(Int) == Float16(0.0)
+    @test Float32(1.1)^unsigned(0) === Float32(1.0)
+    @test Float32(1.1)^big(0) === Float32(1.0)
+
+    # By using a limited-precision integer (3 bits) we can trigger issue 57464
+    # for a case where the answer isn't zero.
```

Comment:
The same is true, if you use for example `Int8` or `Int16`. So why define `Int3` ?